apps/onboarding/lib/view-model.js

```
@@ -247,32 +252,39 @@ export const submitClassFlow = async state => {
       }
       `
     )
-    const { createAppointment: appointment } = await gql.request(
-      `
-      mutation {
-        createAppointment(
-          appointmentId: "${state.selectedAvailability.id}"
-          appointmentTypeId: "${state.apptTypeId}"
-          startTime: "${state.selectedAvailability.startTime}"
-          leadEmail: "${lead.email}"
-        ) {
-          id
-          endTime
-          startTime
-          location {
-            id
-            address1
-            city
-            state
-            zipCode
-            country
+    if (lead.id) {
+      const { createClientFromLead: client } = await submitBilling({
+        ...state,
+        lead
+      })
+      if (client.email) {
+        appointment = (await gql.request(
+          `
+          mutation {
+            createAppointment(
+              appointmentId: "${state.selectedAvailability.id}"
+              appointmentTypeId: "${state.apptTypeId}"
+              startTime: "${state.selectedAvailability.startTime}"
+              leadEmail: "${lead.email}"
+            ) {
+              id
+              endTime
+              startTime
+              location {
+                id
+                address1
+                city
+                state
+                zipCode
+                country
+              }
+            }
           }
-        }
+        `
+        )).createAppointment
+        window.analytics.track('Class Submitted')
       }
-    `
-    )
-    await submitBilling({ ...state, lead })
-    window.analytics.track('Class Submitted')
+    }
     return nextStep({ ...state, appointment, lead })
```

Given the `if` guards `appointment` here could be undefined if it doesn't succeed. Which, in the case of appointments, would fail to render the confirmation and Add to calendar button. This is a state we don't want to show in this case and would prefer to just let an error bubble up to the UI.